src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrCodeflower.cs

```diff
@@ -248,6 +266,91 @@ private async Task<ForwardFlow> GetLastForwardFlow(string mappingName)
         return new ForwardFlow(lastForwardRepoSha, lastForwardVmrSha);
     }
 
+    protected async Task UpdateDependenciesAndToolset(
+        NativePath sourceRepo,
+        ILocalGitRepo targetRepo,
+        Build? build,
+        string currentVmrSha,
+        bool updateSourceElement,
+        CancellationToken cancellationToken)
+    {
+        string versionDetailsXml = await targetRepo.GetFileFromGitAsync(VersionFiles.VersionDetailsXml)
+            ?? throw new Exception($"Failed to read {VersionFiles.VersionDetailsXml} from {targetRepo.Path}");
```

Comment:
It can be useful to know if the file we are trying to read exist in the repo.

Reply:
I am not sure I follow?

You mean to log that we got the file ok?

Reply:
Yes, the `GetFileFromGitAsync()` method calls the `git show ` command (I think) to locate the file. But, what if the file does not exist? --> we'll get `$"Failed to read {VersionFiles.VersionDetailsXml} from {targetRepo.Path}"`. Which does not say the reason. What about to change the exception message to "File does not exist: ..."?

Reply:
```suggestion
            ?? throw new Exception($"Failed to read {VersionFiles.VersionDetailsXml} from {targetRepo.Path} (file does not exist)");
```